Support for ios simulator

I'm trying to build OpenBLAS v0.3.28 for ios simulator with these script and environments:

- macOS 12.7.6
- Xcode 14.2
- Clang 14.0
- Build script:

(`ios-cmake/ios.toolchain.cmake` comes from this repo: https://github.com/leetal/ios-cmake)

```
cmake \
 -B "build" \
 -DDYNAMIC_ARCH=FALSE \
 -DC_LAPACK=ON \
 -DBUILD_TESTING=OFF \
 -DNOFORTRAN=ON \
 -DBUILD_RELAPACK=OFF \
 -DCMAKE_TOOLCHAIN_FILE="ios-cmake/ios.toolchain.cmake" \
 -DENABLE_VISIBILITY=ON \
 -DPLATFORM=SIMULATOR64 \
 -DTARGET=NEHALEM
```

And I encount this error:


```
[ 88%] Building ASM object kernel/CMakeFiles/kernel.dir/CMakeFiles/samax_k.S.o
/Users/momo/Desktop/Lib/OpenBLAS/kernel/x86_64/amax_sse.S:55:2: error: invalid instruction mnemonic 'prologue'
 PROLOGUE
 ^~~~~~~~
/Users/momo/Desktop/Lib/OpenBLAS/kernel/x86_64/amax_sse.S:56:2: error: invalid instruction mnemonic 'profcode'
 PROFCODE
 ^~~~~~~~
/Users/momo/Desktop/Lib/OpenBLAS/kernel/x86_64/amax_sse.S:475:2: error: invalid instruction mnemonic 'epilogue'
 EPILOGUE
 ^~~~~~~~
make[2]: *** [kernel/CMakeFiles/kernel.dir/CMakeFiles/samax_k.S.o] Error 1
make[1]: *** [kernel/CMakeFiles/kernel.dir/all] Error 2
make: *** [all] Error 2

```

In `samax_k.S` (and many other similar files), `PROLOGUE`, `PROFCODE` and `EPILOGUE` is not defined. This is because in `build/config.h`, the system info is detect as `OS_IOS`, and `commom_x86_64.h` can't handle this definition to generate proper `PROLOGUE` and other mnemonic. 


To solve this, I handly change `#define OS_IOS 1` in generated `config.h` to `#define OS_DARWIN 1`, and It seems to work fine. But I'm not familiar with assamble and not sure if there's any potential problem. I'm still trying to figure out how `config.h` is generated and adding support for it. Thanks a lot for any help or information.

--- a/cmake/system_check.cmake
+++ b/cmake/system_check.cmake
@@ -10,6 +10,10 @@ if (${HOST_OS} STREQUAL "WINDOWS")
   set(HOST_OS WINNT)
 endif ()
 
+if (${HOST_OS} STREQUAL "IOS")
+  set(HOST_OS DARWIN)
+endif ()
+
 if (${HOST_OS} STREQUAL "LINUX")
 # check if we're building natively on Android (TERMUX)
     EXECUTE_PROCESS( COMMAND uname -o COMMAND tr -d '\n' OUTPUT_VARIABLE OPERATING_SYSTEM)

